discovery_engine_core/core/src/engine.rs

```diff
@@ -92,14 +113,63 @@ impl Engine {
             .select(self.stacks.values_mut().collect(), max_documents)
             .map_err(|e| e.into())
     }
+
+    /// Process the feedback about the user spending some time on a document.
+    pub fn time_logged(&mut self, time_logged: &TimeLogged) -> Result<(), Error> {
+        self.ranker.time_logged(time_logged)?;
+
+        ranker_updated(self.stacks.values_mut(), &self.ranker)
+    }
+
+    /// Process the feedback about the user reacting to a document.
+    pub fn user_reacted(&mut self, reaction: &UserReacted) -> Result<(), Error> {
+        let stack = self
+            .stacks
+            .get_mut(&reaction.stack_id)
+            .ok_or(Error::InvalidStackId(reaction.stack_id))?;
+
+        stack.update_relevance(reaction.reaction);
+
+        self.ranker.user_reacted(reaction)?;
+
+        ranker_updated(self.stacks.values_mut(), &self.ranker)
+    }
+}
+
+/// The ranker has been updated.
+///
+/// The ranker could rank the documents in a different order so we update the stacks with it.
+fn ranker_updated<'a, R: Ranker>(
```

Comment:
`update_stacks` is too generic, a stack can be updated for multiple reasons (mab paramters, new document, ranker updated). I named this like that because is what we need to do after the ranker has been updated.

Reply:
i see, maybe `update_stacks_with_updated_ranker()` would be an option, but it's also quite unwieldy. anyone else has a concise name for the function which reflects what the function is doing?

Reply:
`rebuild_stacks`? or `refresh_stacks`? (append `_with_ranker` if you think that's not specific enough, although unwieldy IMO as well)
>I named this like that because is what we need to do after the ranker has been updated.

makes sense, but it would still be helpful to know what the function is doing.

Reply:
I called it `rank_stacks`